app/controllers/assignments/rotation_generators_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Assignments
+  class RotationGeneratorsController < ApplicationController
+    before_action :require_admin_in_roster
+    before_action :initialize_rotation_generator
+
+    def prompt; end
+
+    def perform
+      if @generator.generate
+        flash[:message] = t('.success')
+        redirect_to roster_assignments_path(@roster, date: @generator.start_date)
+      else
+        flash.now[:errors] = @generator.errors.full_messages.to_sentence
+        render :prompt, status: :unprocessable_entity
+      end
+    end
+
+    private
+
+    def initialize_rotation_generator
+      @generator = Assignment::RotationGenerator.new(roster_id: @roster.id, **generate_rotation_params)
+      @generator.start_date ||= @roster.next_rotation_start_date
+      @generator.end_date ||= @generator.start_date + 3.months
```

Comment:
```suggestion
      default_start = @roster.next_rotation_start_date
      @generator = Assignment::RotationGenerator.new(roster_id: @roster.id,
                                                     start_date: default_start,
                                                     end_date: default_start + 3.months,
                                                     user_ids: @roster.users.active.collect(&:id),
                                                     **generate_rotation_params)
```
I keep going back and forth on whether I want to request this or not - I think I'm going to leave it as a _light_ suggestion.

I find it a little weird to apply the user-given parameters and _then_ backfill default values.

It's _almost_ a problem in the fact that you might expect on first read that if a user explicitly submitted the form without dates, the blank values would get overridden with these defaults. The fact that we're only submitting through HTML and that blank values _should_ come in as `''` saves the day.

---

`user_ids` addition further explained below